### ISSUE TITLE
Fail to install  H5Z-ZFP in both my local computer and remote cluster

### ISSUE THREAD

**Issue by lzhimin** [LLNL/H5Z-ZFP#122]
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'PREFIX=/home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/h5z-zfp-1.0.1-7rl4zzh2isrkdlcwynkyz7bolpkqj73h' 'CC=/home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/openmpi-4.1.5-js4vse4qmp7nmvtye7lkcuxfx73wm6ah/bin/mpicc' 'HDF5_HOME=/home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/hdf5-1.14.1-2-x7ebmkcnokmiiewa72aqoct3onzkrdhm' 'ZFP_HOME=/home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/zfp-1.0.0-boieacjuaukpfdtk23idw6pkzphtsfrc' 'FC=' 'all'

6 errors found in build log:
  >> 4     grep: /H5Zzfp_plugin.h: No such file or directory
     5     cd src; make  ZFP_HOME=/home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/zfp-1.0.0-boieacjuaukpfdtk23idw6pkzphtsfrc  PREFIX=/home/sci/zhimin/spack/opt/sp
           ack/linux-centos7-haswell/gcc-4.8.5/h5z-zfp-1.0.1-7rl4zzh2isrkdlcwynkyz7bolpkqj73h all
     6     make[1]: Entering directory `/tmp/zhimin/spack-stage/spack-stage-h5z-zfp-1.0.1-7rl4zzh2isrkdlcwynkyz7bolpkqj73h/spack-src/src'
     7     /home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/openmpi-4.1.5-js4vse4qmp7nmvtye7lkcuxfx73wm6ah/bin/mpicc -c H5Zzfp.c -o H5Zzfp_lib.o -DH5Z_ZFP_AS_LIB -
           fPIC -I. -I/home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/zfp-1.0.0-boieacjuaukpfdtk23idw6pkzphtsfrc/include -I/home/sci/zhimin/spack/opt/spack/linux-
           centos7-haswell/gcc-4.8.5/hdf5-1.14.1-2-x7ebmkcnokmiiewa72aqoct3onzkrdhm
     8     /home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/openmpi-4.1.5-js4vse4qmp7nmvtye7lkcuxfx73wm6ah/bin/mpicc -c H5Zzfp_props.c -o H5Zzfp_props.o -fPIC -I.
           -I/home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/zfp-1.0.0-boieacjuaukpfdtk23idw6pkzphtsfrc/include -I/home/sci/zhimin/spack/opt/spack/linux-centos7-h
           aswell/gcc-4.8.5/hdf5-1.14.1-2-x7ebmkcnokmiiewa72aqoct3onzkrdhm
     9     /home/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/openmpi-4.1.5-js4vse4qmp7nmvtye7lkcuxfx73wm6ah/bin/mpicc -c H5Zzfp.c -o H5Zzfp_plugin.o -fPIC -I. -I/ho
           me/sci/zhimin/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/zfp-1.0.0-boieacjuaukpfdtk23idw6pkzphtsfrc/include -I/home/sci/zhimin/spack/opt/spack/linux-centos7-haswel
           l/gcc-4.8.5/hdf5-1.14.1-2-x7ebmkcnokmiiewa72aqoct3onzkrdhm
  >> 10    H5Zzfp.c:42:23: fatal error: bitstream.h: No such file or directory
     11     #include "bitstream.h"
     12                           ^
  >> 13    H5Zzfp.c:42:23: fatal error: bitstream.h: No such file or directory
     14     #include "bitstream.h"
     15                           ^
     16    compilation terminated.
     17    compilation terminated.
     18    ar cr libh5zzfp.a H5Zzfp_props.o
  >> 19    make[1]: *** [H5Zzfp_lib.o] Error 1
     20    make[1]: *** Waiting for unfinished jobs....
  >> 21    make[1]: *** [H5Zzfp_plugin.o] Error 1
     22    make[1]: Leaving directory `/tmp/zhimin/spack-stage/spack-stage-h5z-zfp-1.0.1-7rl4zzh2isrkdlcwynkyz7bolpkqj73h/spack-src/src'
  >> 23    make: *** [all] Error 2

See build log for details:
  /tmp/zhimin/spack-stage/spack-stage-h5z-zfp-1.0.1-7rl4zzh2isrkdlcwynkyz7bolpkqj73h/spack-build-out.txt

**Comment by lindstro** [comment]
@lzhimin This error has me concerned: `H5Zzfp.c:42:23: fatal error: bitstream.h: No such file or directory      11     #include "bitstream.h"`.  zfp 1.0.0 relocated this header, and it seems that you're using an older version of H5Z-ZFP (1.0.1, presumably) that (I believe) is incompatible with zfp 1.0.0.  Can you try using the `develop` version of H5Z-ZFP with Spack?  Someone really ought to update the H5Z-ZFP Spack package to support H5Z-ZFP 1.1.0.

**Comment by markcmiller86** [comment]
I agree with @lindstro H5Z-ZFP-1.01 is not the correct version to use with ZFP-1.0.0.

You should be using H5Z-ZFP-1.1.0.

You can affect this by adding necessary version constratins to your spack build command.

The spack recipe for H5Z-ZFP probably needs to be updated.

**Comment by lzhimin** [comment]
Thanks @lindstro @markcmiller86 
**spack install h5z-zfp@develop** will address the problem.
